securedrop/journalist.py

```diff
@@ -333,6 +349,162 @@ def edit_account():
+    """Get all defined submission tags"""
+    submission_tags = db_session.query(SubmissionLabelType).all()
+    return submission_tags
+
+
+def get_source_tags(source_id):
+    source = Source.query.filter(Source.id == source_id).first()
+    source_tags = db_session.query(SourceTag, SourceLabelType) \
+                            .join(SourceLabelType) \
+                            .filter(SourceTag.source_id == source.id) \
+                            .group_by(SourceLabelType.label_text).all()
+    return source_tags
+
+
+def get_submission_tags(submission_id):
+    submission = Submission.query.filter(Submission.id == submission_id).first()
+    submission_tags = db_session.query(SubmissionTag, SubmissionLabelType) \
+                            .join(SubmissionLabelType) \
+                            .filter(SubmissionTag.submission_id == submission.id) \
+                            .group_by(SubmissionLabelType.label_text).all()
+    return submission_tags
+
+
+def delete_source_label(label_id):
+    matching_label = SourceLabelType.query.filter(SourceLabelType.id == label_id).all()
+    for label in matching_label:
+        db_session.delete(label)
+    db_session.commit()
+
+
+def delete_submission_label(label_id):
+    matching_label = SubmissionLabelType.query.filter(SubmissionLabelType.id == label_id).all()
+    for label in matching_label:
+        db_session.delete(label)
+    db_session.commit()
+
+
+@app.route('/admin/delete_source_label_type/<int:tag_id>', methods=('POST',))
+@admin_required
+def admin_delete_source_label_type(tag_id):
+    delete_source_label(tag_id)
+    return redirect(url_for('admin_index'))
+
+
+@app.route('/admin/create_source_label_type/', methods=('POST',))
+@admin_required
+def admin_create_source_label_type():
+    text_to_display = request.form['text_to_display']
+    all_labels = [x.label_text for x in get_all_defined_source_tags()]
+    if text_to_display not in all_labels:
+        db_session.add(SourceLabelType(label_text=text_to_display))
+        db_session.commit()
+    else:
+        flash('Tag already exists!', 'source')
+    return redirect(url_for('admin_index'))
+
+
+@app.route('/admin/create_submission_label_type/', methods=('POST',))
+@admin_required
+def admin_create_submission_label_type():
+    text_to_display = request.form['text_to_display']
+    all_labels = [x.label_text for x in get_all_defined_submission_tags()]
+    if text_to_display not in all_labels:
+        db_session.add(SubmissionLabelType(label_text=text_to_display))
+        db_session.commit()
+    else:
+        flash('Tag already exists!', 'submission')
+    return redirect(url_for('admin_index'))
+
+
+@app.route('/admin/delete_submission_label_type/<int:tag_id>', methods=('POST',))
+@admin_required
+def admin_delete_submission_label_type(tag_id):
+    delete_submission_label(tag_id)
+    return redirect(url_for('admin_index'))
+
+
+def create_source_tag(source_id, label_id):
```

Comment:
DRYed things up in commit fe6e1ec. 

Also nice point on the views: I researched this a bit and found the following for Flask:

- Flask's [Pluggable Views](http://flask.pocoo.org/docs/0.12/views/), the Flask version of Django's generic views
- [flask-classy](https://github.com/apiguy/flask-classy): a Flask extension to provide class-based views

I haven't used these in this PR but it's something I'm happy to look into for a subsequent PR